Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.8 as base
+
+WORKDIR /app
+
+ENV PYTHONPATH=${PYTHONPATH}:${PWD} 
+
+COPY ./todo_app /app/todo_app
+COPY ./tests /app/tests
+COPY poetry.lock pyproject.toml gunicorn_starter.sh /app
+
+RUN pip3 install poetry
```

Comment:
We want to do this earlier in the Dockerfile. Currently the Docker build has to re-run this command whenever any file changes, because it will rebuild from somewhere in lines 7-9.

Save time by installing poetry first so Docker can cache that intermediate layer.

Reply:
Similar to the above comment, you could take advantage of more caching by doing steps in this order:
- COPY in your poetry files
- RUN poetry install
- COPY in your other files

So Docker can skip the `poetry install` step if pyproject.toml hasn't changed.